sample/src/main/kotlin/com/slack/circuit/sample/home/HomePresenter.kt

```diff
@@ -116,33 +145,89 @@ class HomeUiFactory @Inject constructor() : Ui.Factory {
 private fun homeUi() =
   ui<HomeScreen.State, HomeScreen.Event> { state, events -> HomeContent(state, events) }
 
+@OptIn(ExperimentalMaterialApi::class)
 @Composable
 fun HomeContent(state: HomeScreen.State, eventSink: (HomeScreen.Event) -> Unit) {
-  Scaffold(
-    modifier = Modifier.navigationBarsPadding().systemBarsPadding().fillMaxWidth(),
-    topBar = {
-      CenterAlignedTopAppBar(
-        title = {
-          Text("Adoptables", fontSize = 22.sp, color = MaterialTheme.colorScheme.onPrimaryContainer)
-        },
-        colors =
-          TopAppBarDefaults.centerAlignedTopAppBarColors(
-            containerColor = MaterialTheme.colorScheme.primaryContainer
+  val modalState =
+    rememberModalBottomSheetState(
+      initialValue =
+        if (state.petListFilterState.showBottomSheet) ModalBottomSheetValue.Expanded
+        else ModalBottomSheetValue.Hidden
+    )
+
+  // Monitor bottom sheet state and emit event whenever the user dismisses the modal
+  LaunchedEffect(modalState) {
+    snapshotFlow { modalState.isVisible }
+      .collect { isVisible ->
+        // Toggle if state says the modal should be visible but the snapshot says it isn't.
+        if (state.petListFilterState.showBottomSheet && !isVisible)
+          eventSink(
+            HomeScreen.Event.PetListFilterEvent(PetListFilterScreen.Event.ToggleAnimalFilter)
           )
-      )
-    },
-    bottomBar = {
-      BottomNavigationBar(selectedIndex = state.homeNavState.index) { index ->
-        eventSink(HomeScreen.Event.HomeEvent(HomeNavScreen.Event.HomeNavEvent(index)))
+      }
+  }
+
+  ModalBottomSheetLayout(
```

Comment:
Can we extract the modal sheet bits to a separate file for better readability? This is getting a little hard to read with all the nesting

Reply:
The thing with ModalBottomSheetLayout is that the lambda that follow's is its `content` so while this all could be extracted I'm not sure how much value it would add. If you'd like it all extracted can we do this in a follow up?

Reply:
followup sounds good 👍

Reply:
Issue created to track, https://github.com/slackhq/circuit/issues/136.